packages/@sanity/desk-tool/src/actions/UnpublishAction.tsx

```diff
@@ -43,7 +42,7 @@ export const UnpublishAction: DocumentActionComponent = ({
     onComplete()
   }, [onComplete, unpublish])
 
-  const dialog: DocumentActionDialogProps | null = useMemo(() => {
+  const dialog = ((): DocumentActionDialogProps | null => {
```

Comment:
I personally think it reads better if it's invoked later on instead of immediately (those parens at the end can be easy to miss), and as an added bonus we might end up skipping invocations due to the early return in between. But yeah, no big deal, feel free to leave it as is.